sequencing-server/src/lib/codegen/CommandTypeCodegen.ts

```diff
@@ -133,38 +133,73 @@ ${doc}
   }) as ${fswCommandName}_IMMEDIATE;
 }
-function ${fswCommandName}_STEP(...args: [{ ${argsWithType
-    .map(arg => arg.name + ': ' + arg.type + `${argumentTypeToVariable(arg.type)}`)
-    .join(',')} }]) {
+function ${fswCommandName}_STEP(...args:
+\t\t|[ ${argsWithType.map(arg => `${convertObjectArgsToPassByPosition(arg.name,arg.type)} ${argumentTypeToVariable(arg.type)}`).join(',')} ]
+\t\t|[{ ${argsWithType.map(arg => arg.name + ': ' + arg.type + `${argumentTypeToVariable(arg.type)}`).join(',')} }]) {
   return CommandStem.new({
     stem: '${fswCommandName}',
-    arguments: sortCommandArguments(args, argumentOrders['${fswCommandName}'])
+    arguments: typeof args[0] === 'object' && !Array.isArray(args[0]) && !(args[0] instanceof Variable) ? sortCommandArguments(args, argumentOrders['${fswCommandName}']) : commandArraysToObj(args, argumentOrders['${fswCommandName}']),
   }) as ${fswCommandName}_STEP;
 }`;
 
   const interfaces = `
-\tinterface ${fswCommandName}_IMMEDIATE extends ImmediateStem<[ [{ ${argsWithType
-    .map(arg => arg.name + ': ' + arg.type)
-    .join(',')} }] ]> {}
-\tinterface ${fswCommandName}_STEP extends CommandStem<[ [{ ${argsWithType
-    .map(arg => arg.name + ': ' + arg.type + `${argumentTypeToVariable(arg.type)}`)
-    .join(',')} }] ]> {}
-\tfunction ${fswCommandName}(...args: [{ ${argsWithType
-    .map(arg => arg.name + ': ' + arg.type)
-    .join(',')} }]) : ${fswCommandName}_IMMEDIATE`;
+\tinterface ${fswCommandName}_IMMEDIATE extends ImmediateStem<[
+\t\t | [ ${argsWithType
+      .map(arg => convertObjectArgsToPassByPosition(arg.name,arg.type))
+    .join(',')}]
+\t\t | [{ ${argsWithType
+      .map(arg => arg.name + ': ' + arg.type)
+      .join(',')} }]]> {}
+\tinterface ${fswCommandName}_STEP extends CommandStem<[
+\t\t | [ ${argsWithType
+      .map(arg => `${convertObjectArgsToPassByPosition(arg.name,arg.type)} ${argumentTypeToVariable(arg.type)}`)
+    .join(',')}]
+\t\t | [{ ${argsWithType
+      .map(arg => arg.name + ': ' + arg.type + `${argumentTypeToVariable(arg.type)}`)
+      .join(',')} }]]> {}
+\tfunction ${fswCommandName}(...args:
+\t\t| [ ${argsWithType
+      .map(arg => convertObjectArgsToPassByPosition(arg.name,arg.type))
+    .join(',')}]
+\t\t| [{ ${argsWithType
+      .map(arg => arg.name + ': ' + arg.type)
+      .join(',')} }]) : ${fswCommandName}_IMMEDIATE`;
 
   return {
     value,
     interfaces,
   };
 }
 
+function convertObjectArgsToPassByPosition(name : string, type: string){
+  name = name.replace(/'/g, '');
+  type = type.replace(/[{;}]/g, (match) => {
+    switch (match) {
+      case '{':
+        return '[';
+      case '}':
+        return ']';
+      case ';':
+        return ',';
+      default:
+        return match;
+    }
+  });
+
+  type = type.replace(/'([^']+)':/g, '$1:')
```

Comment:
Can you add comments explaining what this regex and the above 2 are accomplishing?

Reply:
Sure, I added some js doc to the function and some examples.